### PR TITLE
Update to the latest Phoenix and use a LiveView.

### PR DESCRIPTION
**Pull request by kinyoklion**
This effectively is a re-creation of the project using a new generated Phoenix app. This should make it easier for people to get things running for Phoenix, and also demonstrates the LiveView works.

The LiveVIew currently has a 30 second polling interval, but typically you would want to trigger updating the LiveVIew off some data that has changed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,9 +19,6 @@ git clone https://github.com/launchdarkly/hello-phoenix.git
 
 Install dependencies
 ```
-cd hello-phoenix/assets
```
Comment (kinyoklion) [inline README.md:22]: The new Phoenix template project did not require any NPM stuff. Part of why this is now nearly 15k lines smaller.

File: config/config.exs [hunk 2]
```diff
@@ -26,9 +44,9 @@ config :logger, :console,
...
 
-# LaunchDarkly SDK key
-config :hello_phoenix, ld_sdk_key: System.get_env("LD_SDK_KEY")
-
 # Import environment specific config. This must remain at the bottom
 # of this file so it overrides the configuration defined above.
-import_config "#{Mix.env()}.exs"
+import_config "#{config_env()}.exs"
+
+# LaunchDarkly SDK key
```
Comment (kinyoklion) [inline config/config.exs:51]: This line down was added. Same as the original project. Everything above was generated from the template.

File: lib/hello_phoenix/application.ex [hunk 3]
```diff
@@ -5,15 +5,20 @@ defmodule HelloPhoenix.Application do
 
   use Application
 
+  @impl true
   def start(_type, _args) do
     # Start LaunchDarkly SDK client
-    :ldclient.start_instance(String.to_charlist(Application.get_env(:hello_phoenix, :ld_sdk_key)))
-    # List all child processes to be supervised
+    :ldclient.start_instance(String.to_charlist(Application.get_env(:hello_phoenix, :ld_sdk_key, "")))
```
Comment (kinyoklion) [inline lib/hello_phoenix/application.ex:11]: This line of code was added. The rest of the file is the template.

File: lib/hello_phoenix_web/router.ex [hunk 4]
```diff
@@ -17,11 +18,40 @@ defmodule HelloPhoenixWeb.Router do
     pipe_through :browser
 
     get "/", PageController, :index
-    get "/hello-launchdarkly/:user", HelloLaunchdarklyController, :hellouser
+    live "/hello-launchdarkly/:user", HelloLaunchdarklyLive, :index
```
Comment (kinyoklion) [inline lib/hello_phoenix_web/router.ex:21]: I added this live route. The rest is the template.

File: lib/hello_phoenix_web/templates/layout/root.html.heex [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+      <section class="container">
+        <nav>
+          <ul>
+            <li><a href="https://hexdocs.pm/phoenix/overview.html">Get Started</a></li>
+            <%= if function_exported?(Routes, :live_dashboard_path, 2) do %>
+              <li><%= link "LiveDashboard", to: Routes.live_dashboard_path(@conn, :home) %></li>
+            <% end %>
+          </ul>
+        </nav>
+        <a href="https://launchdarkly.com" class="phx-logo">
```
Comment (kinyoklion) [inline lib/hello_phoenix_web/templates/layout/root.html.heex:23]: Added a launchdarkly logo and link here. Similar to the change in the old project. Again the rest of the file is from the template.

File: lib/hello_phoenix_web/views/hello_launchdarkly_live.ex [hunk 6]
```diff
@@ -0,0 +1,36 @@
+defmodule HelloPhoenixWeb.HelloLaunchdarklyLive do
```
Comment (kinyoklion) [inline lib/hello_phoenix_web/views/hello_launchdarkly_live.ex:1]: This is a new file and is the majority of the actual example.

File: lib/hello_phoenix_web/views/hello_launchdarkly_live.ex [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+  use HelloPhoenixWeb, :live_view
+#  use Phoenix.LiveView
+
+  def render(assigns) do
+    case assigns.alternate do
```
Comment (kinyoklion) [inline lib/hello_phoenix_web/views/hello_launchdarkly_live.ex:5]: The old example had some HTML template files, but I did this inline similar to JSX.

**Review by cwaldren-ld — APPROVED** [review]

**Pull request merged**